### DISCUSSION TITLE
External middleware

### DISCUSSION THREAD

**Discussion by KH93b** [nanoexpress/nanoexpress#458] [General]
I want to use external middleware like hpp. Whether the following code is valid?

import nanoexpress from '../src/nanoexpress.js';
const app = nanoexpress();

const hpp = require('hpp');
app.use(hpp());

or  

const helmet= require('helmet');
app.use(helmet());

**Comment by dalisoft** [comment]
Thank you for choosing nanoexpress.
Both of variant are incorrect as you cannot mix CommonJS and ES Module importing.
About using, yes, you using correct usage with `app.use(middleware(middlewareOptions))`

**Reply by KH93b** [reply]
I copied this incorrectly. The bottom line is, can I use third party middleware in nanoexpress?

**Reply by dalisoft** [reply]
Yes, you can
